test/ecto/migration/runner_test.exs

```diff
@@ -0,0 +1,101 @@
+defmodule Ecto.Migration.RunnerTest do
+  use ExUnit.Case, async: false
+
+  alias Ecto.Migration.Runner
+  alias Ecto.Migration.Table
+  alias Ecto.Migration.Index
+
+  defmodule MockAdapter do
+    def execute_migration(_repo, command) do
+      {:migrated, command}
+    end
+
+    def object_exists?(_repo, {:column, {:products, :name}}), do: true
+    def object_exists?(_repo, _), do: false
+  end
+
+  defmodule MockRepo do
+    def adapter do
+      MockAdapter
+    end
+  end
+
+  setup do
+    {:ok, _} = Runner.start_link(MockRepo)
```

Comment:
But then we put the onus on all the callers to know the pid of the migration runner right?
That would mean we need to modify the Runner & DSL to know and pass along the runner's pid `exists? pid, ...` , `execute(pid, ...)` and `direction(pid, ...)`